### PR TITLE
Use cmdstan canonicaliser to update syntax

### PR DESCRIPTION
**Pull request by andrjohns**
@avehtari It looks like the canonicaliser also made some updates to spacing and adding braces to `if` statements. Let me know if you don't want those changes as well, and I can revert.

New syntax checked for correctness via:
```r
stan_files <- list.files(".",pattern=".stan$",recursive=TRUE)

purrr::walk(stan_files, \(stan_file) {
    cmdstan_mod <- cmdstanr::cmdstan_model(stan_file, compile = FALSE)
    cmdstan_mod$check_syntax()
})
```

### CODE REVIEW

File: demos_rstan/bern.stan [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 }
 model {
   // model block creates the log density to be sampled
-  theta ~ beta(1, 1);          // prior 
-  y ~ bernoulli(theta);        // observation model / likelihood
+  theta ~ beta(1, 1); // prior
+  y ~ bernoulli(theta); // observation model / likelihood
   // the notation using ~ is syntactic sugar for
   //  target += beta_lpdf(theta | 1, 1);   // lpdf for continuous theta
-  //  target += bernoulli_lpmf(y | theta); // lpmf for discrete y
```
Comment (avehtari) [inline demos_rstan/bern.stan:15]: Why this line was removed?

File: demos_rstan/binom2.stan [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 parameters {
-  real<lower=0,upper=1> theta1; // probability of death in group 1
-  real<lower=0,upper=1> theta2; // probability of death in group 2
+  real<lower=0, upper=1> theta1; // probability of death in group 1
+  real<lower=0, upper=1> theta2; // probability of death in group 2
 }
 model {
   // model block creates the log density to be sampled
-  theta1 ~ beta(1, 1);          // prior
-  theta2 ~ beta(1, 1);          // prior
```
Comment (avehtari) [inline demos_rstan/binom2.stan:15]: Why is this line removed?

File: demos_rstan/binomb.stan [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
+  int<lower=0> y; // number of successes
 }
 parameters {
   // sampling is done for the parameters
-  real alpha;                  // logit of probability of success in rang (-Inf,Inf)
+  real alpha; // logit of probability of success in rang (-Inf,Inf)
 }
 transformed parameters {
   // transformed parameters are deterministic transformations of parameters (and data)
-  real theta = inv_logit(alpha); // probability of success in range (0,1)
```
Comment (avehtari) [inline demos_rstan/binomb.stan:13]: Why is this line removed?

**Comment by avehtari** [comment]
Strangely it seems like some lines are being removed?

**Comment by andrjohns** [comment]
> Strangely it seems like some lines are being removed?

Ah sorry, I'd fixed that locally but hadn't pushed the changes. One sec

**Comment by andrjohns** [comment]
All updated
